op.c

```diff
@@ -3444,6 +3444,10 @@ Perl_op_lvalue_flags(pTHX_ OP *o, I32 type, U32 flags)
     case OP_SCALAR:
         op_lvalue(cUNOPo->op_first, OP_ENTERSUB);
         goto nomod;
+
+    case OP_ANONCODE:
+        /* We need to not set OPf_REF on this, because of GH#20384 */
```

Comment:
> Personally I don't like such comments. I would prefer seeing something like: `because <minimal explanation of why>, see also GH#20384)`

Fair; I can expand on the reason in the comment.

> (I also wonder if a test is possible/feasible..)

A test would be possible, but only as an XS test that replicates the behaviour that older CPAN modules would perform in the same situation. For such a small code change in obscure situations like this, I wasn't sure if that would be justified.

Reply:
A better comment is added.